nengo/builder/optimizer.py

```diff
@@ -0,0 +1,969 @@
+"""Operator graph optimizers."""
+
+from collections import defaultdict
+import logging
+import time
+import warnings
+
+import numpy as np
+
+from nengo.builder.neurons import SimNeurons
+from nengo.builder import operator
+from nengo.builder.operator import DotInc, ElementwiseInc, SlicedCopy
+from nengo.builder.signal import Signal
+from nengo.config import SupportRcDefaultsMixin
+from nengo.params import BoolParam, IntParam, RcDefault
+from nengo.utils.compat import zip_longest
+from nengo.utils.graphs import toposort, transitive_closure
+
+logger = logging.getLogger(__name__)
+
+
+class OpMergeOptimizer(SupportRcDefaultsMixin):
```

Comment:
I have no idea, due to my inexperience in structuring large Python projects and will defer this to the next reviewer.